libr/util/pkcs7.c

```diff
@@ -654,16 +661,18 @@ static bool r_pkcs7_parse_spcmessagedigest(SpcDigestInfo *messageDigest, RASN1Ob
 		!object->list.objects[0] || !object->list.objects[1]) {
 		return false;
 	}
-	r_x509_parse_algorithmidentifier (&messageDigest->digestAlgorithm, object->list.objects[0]);
+	if (!r_x509_parse_algorithmidentifier (&messageDigest->digestAlgorithm, object->list.objects[0])) {
+		return false;
+	}
 	RASN1Object *obj1 = object->list.objects[1];
 	messageDigest->digest = r_asn1_create_binary (obj1->sector, obj1->length);
 	return true;
 }
 
 R_API SpcIndirectDataContent *r_pkcs7_parse_spcinfo(RCMS *cms) {
-	if (!cms) {
-		return NULL;
-	}
+	r_return_val_if_fail (cms && !strcmp (
+		cms->signedData.contentInfo.contentType->string,
```

Comment:
Not sure if we want to assert on all those details. Maybe just return null for them. The caller wont have to check for them before calling if we dont want runtime crashes